src/fsharp/DependencyManager.Integration.fs

```diff
@@ -62,9 +62,9 @@ module ReflectionHelper =
 type internal IDependencyManagerProvider =
```

Comment:
Could we split the interface in two? the comment for it just above was initial intent, but AFAIU, if I implement the shape for the extension to be picked up, the events aren't supposed to be implemented and won't be used in any case.

approaches you may consider:

1. maybe just removing the events from `IDependencyManagerProvider` and having them on the concrete "empty shell" implementation `ReflectionDependencyManagerProvider` and make the code depend on that concrete implementation
2. have a separate `IDependencyManagerProviderEvents` interface that only `ReflectionDependencyManagerProvider` implements

the first one seems easier and OK for now since it is only CompileOps and tests consuming those events, always from that implementation.

Reply:
The DependencyManager is what invokes the PackageManager, all you do is write an assembly with a class that has a Key Property, a Name Property and a ResolveDependencies method with one of two sets of arguments.  The interface is not used in making the call to the PM.

The events exist so we can tell the host (I.e the notebook what is going on.  The notebook is really interested in loaded dependencies and packages roots.

I can probably rename them a bit to make what they are clearer.  Probably PackageManagerProvider and IPackageManagerProvider are better names.  I don't think splitting it into two interfaces is terribly interesting.

Along with that renaming DependencyManager .dll to NugetPackageManager is probably a vast improvement.

Do you have the paket implementation somewhere, I want to ensure that it will work with C# in notebooks.

And I still have to figure out a better deployment mechanism for other PackageManagers than copying them into the directory containing the compiler.

Reply:
I also just realized, we return an additional argument, that the original API doesn't, so I need to fix that too.

Reply:
@KevinRansom the paket implementation https://github.com/fsprojects/Paket/pull/3783/files#diff-9246adf61946d483ac8c38436bf83ae2

Reply:
@smoothdeveloper, thanks